### PR TITLE
Add animations to row drag (when released)

### LINKED ISSUE #4775: Provide a way to animate row drag, when it is released
## Is your feature request related to a problem? Please describe.
I want to have animation when grid row is dropped and when not dropped on its target.

## Describe the solution you'd like
In such cases, row stays in its source grid, having an animation that transitions the row drag ghost from its current position to its place in the grid will improve grid UX.

### PR DESCRIPTION
**Pull request by dafo**
Closes #4775   

### Additional information (check all that apply):
 - [ ] Bug fix
 - [x] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code
 - [ ] This PR includes API docs for newly added methods/properties
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/grids/row-drag.directive.ts [hunk 1]
```diff
@@ -70,22 +71,44 @@ export class IgxRowDragDirective extends IgxDragDirective implements OnDestroy {
...
+
+        if (args.hasAnimation) {
+            this.animateOnRelease = true;
+        }
+
+        if (!this._lastDropArea && this.animateOnRelease) {
+            requestAnimationFrame(() => {
+                super.onPointerUp(event);
+                this.dragGhost.addEventListener('transitionend',  () => {
+                    this.onTransitionEnd(null);
```
Comment (Lipata) [inline projects/igniteui-angular/src/lib/grids/row-drag.directive.ts]: You can add this into a seperate method

File: projects/igniteui-angular/src/lib/grids/row-drag.directive.ts [hunk 2]
```diff
@@ -70,22 +71,44 @@ export class IgxRowDragDirective extends IgxDragDirective implements OnDestroy {
...
             this.row.grid.onRowDragEnd.emit(args);
         });
-        this._unsubscribe();
+
+        if (args.hasAnimation) {
+            this.animateOnRelease = true;
+        }
+
+        if (!this._lastDropArea && this.animateOnRelease) {
+            requestAnimationFrame(() => {
```
Comment (Lipata) [inline projects/igniteui-angular/src/lib/grids/row-drag.directive.ts]: Is this necessary?

File: projects/igniteui-angular/src/lib/grids/row-drag.directive.ts [hunk 3]
```diff
@@ -70,22 +71,44 @@ export class IgxRowDragDirective extends IgxDragDirective implements OnDestroy {
...
+                super.onPointerUp(event);
+                this.dragGhost.addEventListener('transitionend',  () => {
```
Comment (Lipata) [inline projects/igniteui-angular/src/lib/grids/row-drag.directive.ts]: You need to remove transitionend event listener

File: projects/igniteui-angular/src/lib/grids/grid-base.component.ts [hunk 4]
```diff
@@ -202,9 +202,13 @@ export interface IGridDataBindable {
 export interface IRowDragEndEventArgs {
     owner: IgxDragDirective;
     dragData: IgxRowComponent<IgxGridBaseComponent & IGridDataBindable>;
+    animation: boolean;
 }
 
-export interface IRowDragStartEventArgs extends IRowDragEndEventArgs, CancelableEventArgs { }
+export interface IRowDragStartEventArgs extends CancelableEventArgs {
```
Comment (kdinev) [inline projects/igniteui-angular/src/lib/grids/grid-base.component.ts]: Can't the `IRowDragEnd` just extend the `IRowDragStart`?

Comment (kdinev) [reply projects/igniteui-angular/src/lib/grids/grid-base.component.ts]: @dafo I didn't realize that it's not cancelable. Disregard my comment.

**Pull request merged**